tests/expectations/metrics/test_map_metric.py

```diff
@@ -334,6 +352,403 @@ def test_pandas_unexpected_rows_complete_result_format(dataframe_for_unexpected_
+        "element_count": 6,
+        "unexpected_count": 2,
+        "unexpected_index_list": [
+            {
+                "pk_1": 3,
+            },
+            {
+                "pk_1": 5,
+            },
+        ],  # Dicts since a column was provided
+        "partial_unexpected_index_list": [
+            {
+                "pk_1": 3,
+            },
+            {
+                "pk_1": 5,
+            },
+        ],  # Dicts since a column was provided
+        "unexpected_percent": 33.33333333333333,
+        "partial_unexpected_list": [3, 10],
+        "unexpected_list": [3, 10],
+        "partial_unexpected_counts": [
+            {"value": 3, "count": 1},
+            {"value": 10, "count": 1},
+        ],
+        "missing_count": 0,
+        "missing_percent": 0.0,
+        "unexpected_percent_total": 33.33333333333333,
+        "unexpected_percent_nonmissing": 33.33333333333333,
+    }
+
+
+def test_pandas_multiple_unexpected_index_column_names_complete_result_format(
+    pandas_dataframe_for_unexpected_rows_with_index: pd.DataFrame,
+):
+    expectationConfiguration = ExpectationConfiguration(
+        expectation_type="expect_column_values_to_be_in_set",
+        kwargs={
+            "column": "numbers_with_duplicates",
+            "value_set": [1, 5, 22],
+            "result_format": {
+                "result_format": "COMPLETE",
+                "unexpected_index_column_names": ["pk_1", "pk_2"],  # Multiple columns
+            },
+        },
+    )
+
+    expectation = ExpectColumnValuesToBeInSet(expectationConfiguration)
+    batch: Batch = Batch(data=pandas_dataframe_for_unexpected_rows_with_index)
+    engine = PandasExecutionEngine()
+    validator = Validator(
+        execution_engine=engine,
+        batches=[
+            batch,
+        ],
+    )
+    result = expectation.validate(validator)
+    assert convert_to_json_serializable(result.result) == {
+        "element_count": 6,
+        "unexpected_count": 2,
+        "unexpected_index_list": [
+            {"pk_1": 3, "pk_2": "three"},
+            {"pk_1": 5, "pk_2": "five"},
+        ],  # Dicts since columns were provided
+        "partial_unexpected_index_list": [
+            {"pk_1": 3, "pk_2": "three"},
+            {"pk_1": 5, "pk_2": "five"},
+        ],  # Dicts since columns were provided
+        "unexpected_percent": 33.33333333333333,
+        "partial_unexpected_list": [3, 10],
+        "unexpected_list": [3, 10],
+        "partial_unexpected_counts": [
+            {"value": 3, "count": 1},
+            {"value": 10, "count": 1},
+        ],
+        "missing_count": 0,
+        "missing_percent": 0.0,
+        "unexpected_percent_total": 33.33333333333333,
+        "unexpected_percent_nonmissing": 33.33333333333333,
+    }
+
+
+def test_pandas_multiple_unexpected_index_column_names_complete_result_format_limit_1(
+    pandas_dataframe_for_unexpected_rows_with_index: pd.DataFrame,
+):
```

Comment:
From https://github.com/great-expectations/great_expectations/pull/6329

2nd test for testing partial_unexpected_count limit